### PR TITLE
Improvements to templates

### PR DESCRIPTION
**Pull request by 315234**

### CODE REVIEW

**Comment by noct** [comment]
Unfortunately .C/.H conflicts with separate syntax highlighting for C language files, and I've yet to discover how to force the correct selection behaviour; currently sublime seems to auto-select C++11 by default. So for now users can set a manual preference (clicking the current syntax allows setting a preference for files of that type).

File: C++11.tmLanguage [hunk 1]
```diff
@@ -546,7 +546,7 @@
 		</dict>
 		<dict>
 			<key>match</key>
-			<string>\btemplate\b\s*</string>
+			<string>\btemplate&lt;(class\s+\w+,?\s*)*&gt;</string>
```
Comment (noct) [inline C++11.tmLanguage]: This breaks syntax highlighting for template forms such as:

``` c++
template<typename t>
template<int n>
```

**Comment by 315234** [comment]
I have added two new commits which address these issues.

Are you certain about .C/.H conflicting with C? From what I understand C should use .c/.h.

**Comment by noct** [comment]
Yeah; unfortunately on Windows where files aren't case-sensitive sublime sadly doesn't seem to enforce case-matching on extensions.

**Pull request merged**